src/rtcan.c

```diff
@@ -209,6 +189,54 @@ rtcan_status_t rtcan_init(rtcan_handle_t* rtcan_h,
  */
 rtcan_status_t rtcan_start(rtcan_handle_t* rtcan_h)
 {
+    /* Since the subscriber ID list is known, we can configure the CAN ID Filter now */
+
+    uint32_t can_id_list[RTCAN_HASHMAP_SIZE] = {0};
+    int number_ids = 0;
+
+    /* Save and count all CAN IDs stored in hashmap */
+    for (uint32_t i = 0; i < RTCAN_HASHMAP_SIZE; i++)
+    {
+        if(rtcan_h->subscriber_map[i] != NULL)
+        {
+            /* Subscriber found! */
+            can_id_list[number_ids] = rtcan_h->subscriber_map[i]->can_id;
+            number_ids++;
+
+        }
+    }
+
+    int number_banks = (number_ids-1)/4 + 1;
+
+    /* Error if there's too many ids to apply filter */
+    ADD_ERROR_IF(number_banks > 28, RTCAN_ERROR_INIT, rtcan_h);
+
+    if (no_errors(rtcan_h))
+    {
+        CAN_FilterTypeDef filter;
+        filter.FilterActivation = ENABLE;
+        filter.FilterFIFOAssignment = CAN_FILTER_FIFO0;  
```

Comment:
Do we know what this FIFO assignment does? I don't know if we ever want to use anything other than FIFO0, but maybe its important?

Reply:
According to datasheet: "If there is a match, the message is stored in the associated FIFO". Basically, when there's filter match, the data can be saved either in FIFO0 or FIFO1. Actually, this one may be useful for prioritizing certain IDs over another IDs which are in the same CAN bus.

Reply:
From [slides on the F7 bxCAN peripheral](https://www.st.com/resource/en/product_training/STM32F7_Peripheral_bxCAN.pdf):

> "Two receive FIFOs are used by hardware to store incoming messages. Each FIFO can store three complete messages."

So I don't think there's any inherent prioritisation of one FIFO over another, unless we did this as a feature of RTCAN where it will read and distribute the messages from one FIFO first, but I don't think we need that level of prioritisation.

I think we might want to consider using both FIFOs to share the load (so we use all 6 available message slots rather than just the 3 in FIFO0). Maybe having it alternate between FIFOs each loop?